File: src/main/java/de/kevcodez/ecommerce/parser/downloader/JsoupSourceDownloader.java
package de.kevcodez.ecommerce.parser.downloader;

import java.nio.charset.StandardCharsets;

import org.jsoup.Jsoup;

import lombok.SneakyThrows;

public class JsoupSourceDownloader implements WebsiteSourceDownloader {

    @SneakyThrows
    public String download(String url) {
        return new String(Jsoup.connect(url).execute().bodyAsBytes(), StandardCharsets.UTF_8);
    }

}

File: src/main/kotlin/de/kevcodez/ecommerce/parser/downloader/JsoupSourceDownloader.kt
package de.kevcodez.ecommerce.parser.downloader

import org.jsoup.Jsoup
import java.nio.charset.StandardCharsets

class JsoupSourceDownloader : WebsiteSourceDownloader {

    override fun download(url: String): String {
        return String(Jsoup.connect(url).execute().bodyAsBytes(), StandardCharsets.UTF_8)
    }

}
